### PR TITLE
Use MySQL database

### PR DESCRIPTION
**Pull request by cezres**
- Use MySQL database
- Modify data store of `CellService` (https://github.com/nervosnetwork/ckb-testnet-faucet/pull/85/commits/64d0312c5930e14090576128b5d3f95bb5f5be1e)
- Fixed use `accessToken` to check for vulnerabilities on the date of received (https://github.com/nervosnetwork/ckb-testnet-faucet/pull/85/commits/dccacf52969dc7cf4cc54ed19e1f02e3d3021665)
- Delete api tests and data model tests
- Delete dependency on sqlite

### CODE REVIEW

**Comment by cezres** [comment]
@xiaoluSF  Please note that the database environment parameters have been added.

**Comment by xiaoluSF** [comment]
Merge until cloud MySQL application succeeded

**Comment by ashchan** [comment]
> Merge until cloud MySQL application succeeded

Please re-add me as reviewer when you unfold this.

**Comment by xiaoluSF** [comment]
> > Merge until cloud MySQL application succeeded
> 
> Please re-add me as reviewer when you unfold this.

Done.

File: faucet-server/Sources/App/Controllers/CKBController.swift [hunk 1]
```diff
@@ -28,35 +28,47 @@ public struct CKBController: RouteCollection {
...
-                    let txHash = try sendCapacity(address: address)
-                    Authorization().recordCollectionDate(accessToken: accessToken!)
-                    result = ["status": 0, "txHash": txHash]
-                } else {
-                     result = ["status": -3, "error": "No address"]
+    func faucet(_ req: Request) -> Future<Response> {
+        let urlParameters = req.http.urlString.urlParametersDecode
+        let accessToken = req.http.cookies.all[accessTokenCookieName]?.string ?? ""
+        let email = (try? GithubService.getUserInfo(for: accessToken).email) ?? ""
+        var isSucceed = false
```
Comment (ashchan) [inline faucet-server/Sources/App/Controllers/CKBController.swift]:
```suggestion
        var isSuccess = false
```

Or `isSuccessful`. `Succeed` is a verb.

File: faucet-server/Sources/App/Controllers/CKBController.swift [hunk 2]
```diff
@@ -28,35 +28,47 @@ public struct CKBController: RouteCollection {
...
+        return Authentication().verify(email: email, on: req).map { status -> String in
+            // Send capacity
+            if status == .tokenIsVailable {
+                do {
+                    if let address = urlParameters["address"] {
+                        txHash = try self.sendCapacity(address: address)
+                        isSucceed = true
+                        return ["status": 0, "txHash": txHash].toJson
+                    } else {
+                        return ["status": -3, "error": "No address"].toJson
```
Comment (ashchan) [inline faucet-server/Sources/App/Controllers/CKBController.swift]: `-3`, `-4` and `0` are magic numbers. How about defining an enum of Int?

File: faucet-server/Sources/App/Extensions/EnvironmentExtensions.swift [hunk 3]
```diff
@@ -9,18 +9,33 @@ import Vapor
...
         static var nodeURL: String!
-        static var oauthClientId: String!
-        static var oauthClientSecret: String!
         static var walletPrivateKey: String!
         static var sendCapacityCount: Decimal!
+        // oauth
+        static var oauthClientId: String!
+        static var oauthClientSecret: String!
+        // db
+        static var dbHostname: String!
```
Comment (ashchan) [inline faucet-server/Sources/App/Extensions/EnvironmentExtensions.swift]: As configuration becomes complicated, perhaps it's time we grouped related properties into sub structures. For example, putting db properties inside `Environment.Process.db`.

File: faucet-server/Sources/App/Models/Faucet.swift [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+//
+//  Created by 翟泉 on 2019/5/22.
+//
+
+import Foundation
+import Vapor
+import Fluent
+import FluentMySQL
+
+struct Faucet: Content, MySQLModel {
```
Comment (ashchan) [inline faucet-server/Sources/App/Models/Faucet.swift]: I have difficult understand what a faucet model is. Is it to store records of actual capacity sending out to a GitHub user (by using email as their identity)?

File: faucet-server/Sources/App/Services/Authentication.swift [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+    func verify(email: String, on connection: Request) -> EventLoopFuture<Status> {
+        return User
+            .query(on: connection)
+            .filter(\.email, .equal, email)
+            .first()
+            .map { user -> Status in
+            if let user = user {
+                if user.recentlyReceivedDate?.timeIntervalSince1970 ?? 0 < Date().timeIntervalSince1970 - 24 * 60 * 60 {
+                    return .tokenIsVailable
```
Comment (ashchan) [inline faucet-server/Sources/App/Services/Authentication.swift]: `tokenIsVailable` seems to be a name having typo.

File: faucet-server/Sources/App/Extensions/EnvironmentExtensions.swift (hunk 3 above)
Comment (cezres) [reply faucet-server/Sources/App/Extensions/EnvironmentExtensions.swift]: https://github.com/nervosnetwork/ckb-testnet-faucet/pull/85/commits/42e1e0ab8146791510e2a07b9b01843320484d3a

File: faucet-server/Sources/App/Controllers/CKBController.swift (hunk 2 above)
Comment (cezres) [reply faucet-server/Sources/App/Controllers/CKBController.swift]: https://github.com/nervosnetwork/ckb-testnet-faucet/pull/85/commits/28e042d709ada331068461571cea010bb113037e

File: faucet-server/Sources/App/Services/Authentication.swift (hunk 5 above)
Comment (cezres) [reply faucet-server/Sources/App/Services/Authentication.swift]: https://github.com/nervosnetwork/ckb-testnet-faucet/pull/85/commits/28e042d709ada331068461571cea010bb113037e

File: faucet-server/Sources/App/Models/Faucet.swift (hunk 4 above)
Comment (cezres) [reply faucet-server/Sources/App/Models/Faucet.swift]: It a log of the user receiving CKB.

**Pull request merged**